File: Vishwas RNO 40/DotNet/Assignment 4/DemoApp/series/PowerSequence.cs
namespace series;

class PowerSequence : Sequence , IResetable
{
    private double current;
    private double factor;

    public PowerSequence(double factor)
    {
        this.factor = factor;
        current = 1;
    }

    public override double Next()
    {
        double term = current;
        current *= factor;
        return term;
    }

    public void Reset()
    {
        current = 1;
    }
} 

File: Vishwas RNO 40/Java/Assign5/src/PowerSequence.java
package series;

public class PowerSequence extends Sequence implements Resetable {

	private double current;
	private float factor;

	public PowerSequence(float ratio) {
		current = 1;
		factor = ratio;
	}

	public double next() {
		
		double term = current;
		current *= factor;
		return term;
	}

	public void reset() {
		current = 1;
	}

		

}
